File: LabTFT/c_files/code.c
#include <stdio.h>
#include <time.h>
#include "lib.h"
#include "colors.h"

Font Font5x7_struct = { Font5x7, 5, 7 };



int main() {
    printf("Call to config_pins\n");

    init();

    printf("Call to fill_screen:\n");

    struct timespec start, end;

    clock_gettime(CLOCK_MONOTONIC, &start);

    clear_screen(ORANGE);

    clock_gettime(CLOCK_MONOTONIC, &end);

    printf("Pantalla limpia, ejecutando draw_text_scaled...\n");
//    draw_image_file(0,0,Size_X,Size_Y,"../images/on.png");
    draw_text_scaled(10, 10, "HOLA MUNDO :)", RED, Font5x7_struct, 2);

    double elapsed = (end.tv_sec - start.tv_sec) + (end.tv_nsec - start.tv_nsec) / 1e9;

    printf("Temps: %.6f segons\n", elapsed);

    return 0;
}

File: Entrega2/Lab 5 - Pantalla TFT/python_files/code.py
import lib
import time
import colors

if __name__ == "__main__":
    try:
        lib.init()

        # Pintar pantalla completa de verd i imprimir temps transcorregut
        print("Pintant pantalla completa...")
        start = time.time()
        lib.clear_screen(colors.GREEN)
        end = time.time()

        duration = end - start
        print(f"Ha trigat {duration:.6f} segons a pintar la pantalla completa.")

        time.sleep(5)

    finally:
        lib.quit()
